keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -189,12 +190,18 @@ def __init__(
             include_preprocessing=preprocessor is not None,
             **kwargs,
         )
-
         self.backbone = backbone
         self.preprocessor = preprocessor
         self.generate_function = None
-        # Private sampler set by compile.
-        self._sampler = samplers.get("top_k")
+        self._sampler = None
+
+        # Default compilation
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(2e-5),
```

Comment:
Random note -- I thought AdamW was preferred?

Reply:
We have so far avoided it as for tf 2.9 adamw is very broken (does not converge with no error messages). But now that 2.9 is no longer the default version on colab, we could consider using it everywhere.